include/net/can.h

```diff
@@ -131,6 +131,14 @@ struct canbus_net_ctx {
 	struct net_if *iface;
 	/** The link layer address chosen for this interface */
 	uint16_t ll_addr;
+	/** TX queue */
+	struct k_fifo tx_queue;
+	/** RX error queue */
+	struct k_fifo rx_err_queue;
+	/** Queue handler thread */
+	struct k_thread queue_handler;
+	/** Queue handler thread stack */
+	K_KERNEL_STACK_MEMBER(queue_stack, 512);
```

Comment:
It looks like "600 bytes * number of CAN contexts" more RAM is used after this change. I am fine with this change as is, but we might want to use in the future single thread for CAN, same as for PPP.

Reply:
> It looks like "600 bytes * number of CAN contexts" more RAM is used after this change. I am fine with this change as is, but we might want to use in the future single thread for CAN, same as for PPP.

Yeah, the 6locan driver is almost supporting multiple contexts / interfaces. It looks like the purpose was to support multiple context but it there are some bits and pieces missing to fully support this. I did not want to start to tweak this driver for now, but it can be done later if really needed. I will leave this to @alexanderwachter to decide.